### PR TITLE
Document additional credential providers for JDBC connectors

### LINKED ISSUE #1910: Document use of extra credentials in JDBC connectors
release notes:
```
* Add support for using a client-provided username and password. The credential
  names can be configured using the ``user-credential-name`` and ``password-credential-name``
  configuration properties. (:issue:`482`)
```

we should have proper documentation for this

related to https://github.com/prestosql/presto/issues/1911

### PR DESCRIPTION
**Pull request by nineinchnick**
## Description

Document credential providers introduced in #1124 and #1521

## Non-technical explanation

n/a

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

Fixes #1910, #1911

### CODE REVIEW

**Comment by nineinchnick** [comment]
@Jessie212 @mosabua maybe you could take a look at this one

File: docs/src/main/sphinx/connector/jdbc-authentication.fragment [hunk 1]
```diff
@@ -0,0 +1,55 @@
+Data source authentication
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The connector can provide credentials for the data source connection
+in multiple ways:
```
Comment (Jessie212) [inline docs/src/main/sphinx/connector/jdbc-authentication.fragment:5]: Maybe say ... in the following ways:

Imo, "following" anticipates a list and explanation more than simply saying "multiple", which sounds like a stop.

File: docs/src/main/sphinx/connector/jdbc-authentication.fragment [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+* inline, in the connector configuration file
+* in a separate properties file
+* in a key store file
+* as extra credentials set when connecting to Trino
```
Comment (Jessie212) [inline docs/src/main/sphinx/connector/jdbc-authentication.fragment:10]: Capitalize the first letter in each bullet point and place a period at the end.

**Comment by Jessie212** [comment]
I think this is a good addition to the docs.

File: docs/src/main/sphinx/connector/jdbc-authentication.fragment (hunk 1 above)
Comment (nineinchnick) [reply docs/src/main/sphinx/connector/jdbc-authentication.fragment]: I like that, thanks!

File: docs/src/main/sphinx/connector/jdbc-authentication.fragment (hunk 2 above)
Comment (nineinchnick) [reply docs/src/main/sphinx/connector/jdbc-authentication.fragment]: Done!

File: docs/src/main/sphinx/connector/jdbc-authentication.fragment [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    - Connection user name.
+  * - ``connection-password``
+    - Connection password.
+  * - ``user-credential-name``
+    - Name of the extra credentials property, which value to use
+      as the user name. See ``extraCredentials``
+      in :ref:`jdbc-parameter-reference`.
+  * - ``password-credential-name``
+    - Name of the extra credentials property, which value to use
+      as the password.
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/jdbc-authentication.fragment:37]:
```suggestion
    - Name of the extra credentials property, whose value to use
      as the user name. See ``extraCredentials``
      in :ref:`jdbc-parameter-reference`.
  * - ``password-credential-name``
    - Name of the extra credentials property, whose value to use
      as the password.
```

**Review by hashhar — APPROVED** [review]

File: docs/src/main/sphinx/connector/jdbc-authentication.fragment (hunk 3 above)
Comment (nineinchnick) [reply docs/src/main/sphinx/connector/jdbc-authentication.fragment:37]: @mosabua @Jessie212 WDYT? I had to google this and couldn't find a definitive answer.

**Comment by nineinchnick** [comment]
This supersedes #11791

File: docs/src/main/sphinx/connector/jdbc-authentication.fragment (hunk 3 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/jdbc-authentication.fragment:37]: That change is correct.

Comment (hashhar) [reply docs/src/main/sphinx/connector/jdbc-authentication.fragment:37]: That = current, or suggested?

Comment (Jessie212) [reply docs/src/main/sphinx/connector/jdbc-authentication.fragment:37]: It sounds better to say: 

... whose value is used as the username.
... whose value is used as the password.

Comment (nineinchnick) [reply docs/src/main/sphinx/connector/jdbc-authentication.fragment]: Thanks, updated!

File: docs/src/main/sphinx/connector/jdbc-authentication.fragment [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  * - ``keystore-type``
+    - File format of the keystore file, for example ``JKS`` or ``PEM``.
+  * - ``keystore-password``
+    - Password for the key store.
+  * - ``keystore-user-credential-name``
+    - Name of the key store entity to be used as the user name.
+  * - ``keystore-user-credential-password``
+    - Password for the user name key store entity.
+  * - ``keystore-password-credential-name``
+    - Name of the key store entity to be used as the password.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/jdbc-authentication.fragment:51]:
```suggestion
    - Name of the key store entity to use as the password.
```

File: docs/src/main/sphinx/connector/jdbc-authentication.fragment [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    - Location of the properties file where credentials are present. It should
+      contain the ``connection-user`` and ``connection-password`` properties.
+  * - ``keystore-file-path``
+    - The location of the Java Keystore file, from which to read credentials.
+  * - ``keystore-type``
+    - File format of the keystore file, for example ``JKS`` or ``PEM``.
+  * - ``keystore-password``
+    - Password for the key store.
+  * - ``keystore-user-credential-name``
+    - Name of the key store entity to be used as the user name.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/jdbc-authentication.fragment:47]:
```suggestion
    - Name of the key store entity to use as the user name.
```

File: docs/src/main/sphinx/connector/jdbc-authentication.fragment [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  * - ``connection-password``
+    - Connection password.
+  * - ``user-credential-name``
+    - Name of the extra credentials property, whose value to use as the user
+      name. See ``extraCredentials`` in :ref:`jdbc-parameter-reference`.
+  * - ``password-credential-name``
+    - Name of the extra credentials property, whose value to use as the
+      password.
+  * - ``connection-credential-file``
+    - Location of the properties file where credentials are present. It should
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/jdbc-authentication.fragment:38]:
```suggestion
    - Location of the properties file where credentials are present. It must
```

**Review by mosabua — CHANGES_REQUESTED** [review]
Some minor improvements needed.. then good to go.

**Review by hashhar — APPROVED** [review]
LGTM from me.

@mosabua please approve if it looks good to you as well and I'll merge.

**Review by mosabua — APPROVED** [review]
Good to go!

**Comment by ebyhr** [comment]
Please note that `Fixes #1910, #1911` doesn't close the 2nd issue. We could add another `Fixes` like `Fixes #1910, Fixes #1911`

**Pull request merged**